idaes/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/hetero_reactions.py

```diff
@@ -0,0 +1,517 @@
+                             initialize=1.0,
+                             doc='Reaction order in gas species [-]')
+        self.rxn_order.fix()
+
+        # Pre-exponential factor
+        self.k0_rxn = Var(self.rate_reaction_idx,
+                          domain=Reals,
+                          initialize=3.1e-4,
+                          doc='Pre-exponential factor'
+                          '[mol^(1-N_reaction)m^(3*N_reaction -2)/s]')
+        self.k0_rxn.fix()
+
+    @classmethod
+    def define_metadata(cls, obj):
+        obj.add_properties({
+                'k_rxn': {'method': '_k_rxn',
+                          'units': 'mol^(1-N_reaction)m^(3*N_reaction -2)/s]'},
+                'OC_conv': {'method': "_OC_conv", 'units': None},
+                'OC_conv_temp': {'method': "_OC_conv_temp", 'units': None},
+                'reaction_rate': {'method': "_reaction_rate",
+                                  'units': 'mol_rxn/m3.s'}
+                })
+        obj.add_default_units({'time': 's',
+                               'length': 'm',
+                               'mass': 'kg',
+                               'amount': 'mol',
+                               'temperature': 'K',
+                               'energy': 'kJ'})
+
+
+class _ReactionBlock(ReactionBlockBase):
+    """
+    This Class contains methods which should be applied to Reaction Blocks as a
+    whole, rather than individual elements of indexed Reaction Blocks.
+    """
+    def initialize(blk, outlvl=idaeslog.NOTSET,
+                   optarg={'tol': 1e-8}, solver='ipopt'):
+        '''
+        Initialisation routine for reaction package.
+
+        Keyword Arguments:
+            outlvl : sets output level of initialization routine
+                 * 0 = Use default idaes.init logger setting
+                 * 1 = Maximum output
+                 * 2 = Include solver output
+                 * 3 = Return solver state for each step in subroutines
+                 * 4 = Return solver state for each step in routine
+                 * 5 = Final initialization status and exceptions
+                 * 6 = No output
+            optarg : solver options dictionary object (default=None)
+            solver : str indicating whcih solver to use during
+                     initialization (default = "ipopt")
+        Returns:
+            None
+        '''
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="reactions")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="reactions")
+
+        init_log.info_high('Starting initialization')
+
+        # TODO - Update in the future as needed
+        # Get a single representative block for getting config arguments
+        for k in blk.keys():
+            break
+
+        # Fix state variables if not already fixed
+        # Fix state variables of the primary (solid) state block
+        state_var_flags = fix_state_vars(blk[k].config.solid_state_block)
```

Comment:
This maybe required here, but we should be careful not to get ourselves in a mess with different things fixing and unfixing state variables. For this case, I would suggest (if possible) that the unit model tells the state block to fix and hold the state variables, then the unit model calls the reaction initialization and finally the unit model tells the state block to unfix the state variables.

I.e. Let the reaction block assume the state is fixed (and initialized) already. At most, maybe the reaction block should check that the state variables are fixed and raise an exception if otherwise?

The reason for this is just to make things easier to follow.

Reply:
I think I agree with this, but the reaction block requires more variables than just the states to be fixed, right? So you're drawing a distinction between state variables and other variables, where the unit model fixes state vars, then the reaction block does whatever it needs to during initialization to get a square problem, assuming state variables are fixed?